presto-main/src/test/java/com/facebook/presto/operator/scalar/TestArrayConstructor.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.scalar;
+
+import com.facebook.presto.type.ArrayType;
+import org.testng.annotations.Test;
+
+import java.util.stream.IntStream;
+
+import static com.facebook.presto.spi.type.DoubleType.DOUBLE;
+import static com.google.common.base.Verify.verify;
+import static org.testng.Assert.assertTrue;
+
+public class TestArrayConstructor
+        extends AbstractTestFunctions
+{
+    @Test
+    public void testArgumentCount()
+    {
+        tryEvaluateWithAll(simpleArrayConstructionOf(254), new ArrayType(DOUBLE));
+
+        try {
+            tryEvaluateWithAll(simpleArrayConstructionOf(255), new ArrayType(DOUBLE));
+        }
+        catch (IllegalArgumentException e) {
```

Comment:
We should also have an error message that is meaningful to an end user (rather than specific to the internals of bytecode generation)